JavaScript/JavaScriptSDK/TelemetryContext.ts

```diff
@@ -17,6 +17,8 @@ module Microsoft.ApplicationInsights {
         accountId: () => string;
         sessionRenewalMs: () => number;
         sessionExpirationMs: () => number;
+        properties: () => Object;
+        measurements: () => Object;
```

Comment:
Could you elaborate why you've chosen this particular design and its tradeoffs compared to the available alternatives? I've listed some of the alternative designs that have been suggested or implemented previously.

- This is a direct port of the .NET interface, similar to the 'onInitialize'/'onBeforeSendTelemetry` suggestion made by @SergeyKanzhelev and @AlexBulankou.
```TypeScript
export interface ITelemetryInitializer {
    initialize(envelope: Telemetry.Common.Envelope): void;
}
```

- A single-value initializer, where user-defined function returns value of a single property and is registered like this:
```TypeScript
config.properties["foo_property"] = () => "bar_value";
```